### PR TITLE
fix(grid): do not escape edit mode on api update #2773

### LINKED ISSUE #2773: Template editor does not end edit by calling cell.update() on Enter pressed
### Description  
Template editor does not end edit by calling cell.update() on Enter key press.

 * igniteui-angular version: 6.2.0-beta.3
Note: This is not reproduced on 6.0.3

 * browser: Chrome, IE11

### Steps to reproduce  

1. Run the attached sample
2. Enter edit mode on "startDate" cell
3. Type some characters on the editor
4. Hit enter to end edit mode

### Result  
Edit mode does not end.

### Expected result  
Edit mode ends and edited value is set to the cell.

### Attachments
[igniteui-app1.zip](https://github.com/IgniteUI/igniteui-angular/files/2494483/igniteui-app1.zip)

### PR DESCRIPTION
**Pull request by hanastasov**
Closes #2773

Additional information related to this pull request:
- API update method no longer escapes edit mode in consistency with all other scenarios when cell is not exiting edit mode
- this results in only 1 call of escape_edit_mode, test expectation changed accordingly
- manually updating the cell.editValue when updating through the API

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/grid/grid.component.spec.ts [hunk 1]
```diff
@@ -2144,8 +2144,7 @@ describe('IgxGrid Component Tests', () => {
                 expect(cell.inEditMode).toBe(false);
                 expect(cell.value).toBe(110); // SORT does not submit
 
-                expect(gridAPI.escape_editMode).toHaveBeenCalled();
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/grids/grid/grid.component.spec.ts]: expect(gridAPI.escape_editMode).toHaveBeenCalled(); should exist in the test.

Comment (hanastasov) [reply projects/igniteui-angular/src/lib/grids/grid/grid.component.spec.ts]: It exists on next line doesnt it

**Comment by zdrawku** [comment]
@ddincheva could you check this PR as well

**Comment by kdinev** [comment]
@hanastasov Has this been merged into `6.2.x`? If yes, then it will make it into master with a mass merge from 6.2.x.

**Comment by hanastasov** [comment]
We can close this PR and later it will be merged to 6.2.x (after implementing changes requested). Is that ok ?

**Comment by hanastasov** [comment]
Closing this PR, changes will be introduced to the [PR in 6.2.x](https://github.com/IgniteUI/igniteui-angular/pull/2969) and then will make it into master with mass merge.

**Comment by hanastasov** [comment]
6.2.x has already been merged to master, so reopening this PR.

**Comment by hanastasov** [comment]
TO DECIDE: This PR changes the behavior when a cell is in edit mode and cell/row is updated via API. I am going to push anther fix, if cell needs to exit edit mode in those situations.

**Comment by kdinev** [comment]
There's a new PR to even 6.2.x and master so closing this again.

**Pull request closed without merging**